apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/AbstractCleanupJobTest.java

```diff
@@ -0,0 +1,128 @@
+package gov.cms.bfd.pipeline.rda.grpc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import gov.cms.bfd.pipeline.CleanupTestUtils;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/** Test class for {@link gov.cms.bfd.pipeline.rda.grpc.AbstractCleanupJob}. */
+class AbstractCleanupJobTest {
+
+  /** test utilities. */
+  private CleanupTestUtils utils;
+
+  /** test initialization. */
+  @BeforeEach
+  void init() {
+    utils = new CleanupTestUtils();
+    utils.init();
+  }
+
+  /** test teardown. */
+  @AfterEach
+  void cleanupAfterEach() {
+    utils.truncateTables();
+  }
+
+  /**
+   * Basic test scenario, normal case where it executes multiple transactions to remove old claims.
+   *
+   * @throws ProcessingException if errors occurs during processing.
+   */
+  @Test
+  void runComplete() throws ProcessingException {
+    var cutoff = Instant.now().minus(60, ChronoUnit.DAYS);
+    utils.seedData(cutoff, 8, 4);
+
+    var cleanUpJob = new RdaFissClaimCleanupJob(utils.getTransactionManager(), 6, 2, true);
+    var deleted = cleanUpJob.run();
+
+    // after first run 6 should be deleted and 6 remaining
+    assertEquals(6, deleted);
+
+    deleted = cleanUpJob.run();
+
+    // after second run 2 should be deleted and 4 remaining, none older than 60 days;
+    assertEquals(2, deleted);
+    assertEquals(4, utils.count());
```

Comment:
Maybe run it again and verify 0 are deleted?

Reply:
Done